### PR TITLE
migrate connection in Google Cloud BigQuery and Spanner Tests

### PR DESCRIPTION
**Pull request by itsankit-google**
[PLUGIN-987](https://cdap.atlassian.net/browse/PLUGIN-987)

### CODE REVIEW

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/DataprocETLTestBase.java [hunk 1]
```diff
@@ -177,4 +180,52 @@ private JsonObject ofProperty(String name, String value) {
...
+    JsonObject plugin = new JsonObject();
+    plugin.addProperty("category", "Google Cloud Platform");
+    plugin.addProperty("name", connectorName);
+    plugin.addProperty("type", "connector");
+    plugin.add("properties", properties);
+    plugin.add("artifact", artifact);
+
+    JsonObject body = new JsonObject();
+    body.addProperty("name", connectionName);
+    body.addProperty("description", "Connection for running bq_integration_test");
```
Comment (yaojiefeng) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/DataprocETLTestBase.java]: this is in test base, please update the description

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/DataprocETLTestBase.java [hunk 2]
```diff
@@ -177,4 +180,52 @@ private JsonObject ofProperty(String name, String value) {
     jsonObject.addProperty("value", value);
     return jsonObject;
   }
+
+  protected void createConnection(String connectionName, String connectorName) throws Exception {
```
Comment (yaojiefeng) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/DataprocETLTestBase.java:185]: I think we will need methods for CRUD operations for connection

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java [hunk 3]
```diff
@@ -238,7 +240,8 @@ public void testReadDataAndStoreInNewTable() throws Exception {
    */
   private void testReadDataAndStoreInNewTable(Engine engine) throws Exception {
     String testId = GoogleBigQueryUtils.getUUID();
-
+    createConnection(CONNECTION_NAME, "BigQuery");
```
Comment (yaojiefeng) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java]: In ITN, it is important to remove the resources after each test is run to make sure each test is isolated. You can see that in each test the profile is created before each test and removed after each test. And all the pipelines will get deleted after each test. Please make sure this also applies to connections. After each test, we should list the connections, and remove the connections.

Comment (itsankit-google) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java]: acknowledged.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/DataprocETLTestBase.java (hunk 1 above)
Comment (itsankit-google) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/DataprocETLTestBase.java]: acknowledged.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/DataprocETLTestBase.java (hunk 2 above)
Comment (itsankit-google) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/DataprocETLTestBase.java:185]: For now, looks like there is no Api for connection updation, will add a connection deletion method.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java [hunk 4]
```diff
@@ -251,15 +255,17 @@ private void testReadDataAndStoreInNewTable(Engine engine) throws Exception {
 
     Map<String, String> sourceProps = new ImmutableMap.Builder<String, String>()
       .put("referenceName", "bigQuery_source")
-      .put("project", "${project}")
+      .put(ConfigUtil.NAME_CONNECTION, connectionId)
```
Comment (yaojiefeng) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java]: can we make sure the tests are also run using the original config without connection? Like we discussed today, we should introduce some flag to control whether to use connection or not in each test

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java [hunk 5]
```diff
@@ -207,16 +209,54 @@ protected void innerSetup() throws Exception {
...
+    }
+    sourceProps.put("dataset", "${dataset}");
+    sourceProps.put("table", "${srcTable}");
+    if (sourceSchema != null) {
+      sourceProps.put("schema", sourceSchema.toString());
+    }
+    return new ImmutableMap.Builder<String, String>().putAll(sourceProps).build();
+  }
+
+  private Map<String, String> getSinkProps(Boolean useConnection) {
```
Comment (yaojiefeng) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java]: this method and above method seems very similar, can we combine these into one method. Also can just use `boolean` here

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleCloudSpannerTest.java [hunk 6]
```diff
@@ -215,40 +218,72 @@ public void testSetup() throws Exception {
 
     // Insert test data
     spanner.getDatabaseClient(database.getId()).write(SOURCE_TABLE_TEST_MUTATIONS);
+    createConnection(CONNECTION_NAME, "Spanner");
+  }
+
+  private Map<String, String> getSourceProps(Boolean useConnection, Schema sourceSchema) {
```
Comment (yaojiefeng) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleCloudSpannerTest.java]: same here, can these two methods combined into one?

Comment (itsankit-google) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleCloudSpannerTest.java]: acknowledged.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java (hunk 5 above)
Comment (itsankit-google) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/gcp/GoogleBigQueryTest.java]: acknowledged.

**Review by yaojiefeng — APPROVED** [review]

**Pull request merged**